File: GraspItApi/Models/Deck.cs
public class Deck
{
    public int DeckId {get; set;}
    public int? UserId {get; set;}
    public User? User{get; set;}
    public List<Card>? Cards {get; set;}
    public required string Name {get; set;}
    public required string Description {get; set;}
} 

File: GraspItApi/src/main/java/com/devkamogelo/graspit/models/Deck.java
package com.devkamogelo.graspit.models;

import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.List;
import java.util.UUID;

@Data
@Entity
@Table(name = "decks")
public class Deck {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @OneToMany(mappedBy = "deck", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    private List<Card> cards;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "user_id")
    private User user;
}
